stdlib/logging/config.pyi

```diff
@@ -15,9 +16,48 @@ if sys.version_info >= (3, 7):
 else:
     _Path = StrPath
 
+_TKey = TypeVar("_TKey")
+_TValue = TypeVar("_TValue")
+
 DEFAULT_LOGGING_CONFIG_PORT: int
 RESET_ERROR: int  # undocumented
 IDENTIFIER: Pattern[str]  # undocumented
+dictConfigClass = DictConfigurator
+
+class ConvertingMixin(Generic[_TKey, _TValue]):
+    def convert_with_key(self, key: _TKey, value: _TValue, replace: bool = ...) -> Dict[_TKey, _TValue]: ...
```

Comment:
I don't entirely understand what's going on in this code but it doesn't look like it always returns a dict. Some of the other stubs below instead assume that it returns `_TValue`. Conceptually I'd think there should be separate input and output value types.

I'm hesitant to add these classes to our stubs since they're undocumented and it's difficult to get the types right. From your PR description it sounds like you started this PR to better type config dicts; it's not clear to me that doing so requires adding all of these classes to the stubs.

A lot of the Dicts in this module can probably be replaced with `TypedDict`—that may be the best way to improve type safety for users.